### PR TITLE
chore: set poetry up

### LINKED ISSUE #30: Set poetry up
- Set poetry up for project
- Add tasks using `taskipy`
- Change contributing instructions based on changes

### PR DESCRIPTION
**Pull request by DudeBro249**
- Set poetry up for project and fixed imports
- Added lint and bot tasks with `taskipy`
- Changed some development instructions to reflect poetry setup
- Changed `Dockerfile` to use poetry
- Changed `linting.yml` worfklow to use poetry
- Closes #30

### CODE REVIEW

**Comment by GDWR** [comment]
Do you know how/where the venv is stored for poetry. The reason I didn't do this straight away is that Pycharm Docker Interpreters for Python allows you to target a Python binary within the container to use. Using Poetry will add some complications to this that I didn't want to find a workaround for.

If you let me know where the venv is within the container, I'll try and use it with a docker interpreter to see if this would be okay.

**Comment by DudeBro249** [comment]
I've added a file called `poetry.toml` which stores some extra poetry configuration such as whether or not to create a virtualenv and whether or not to place it in the root directory. This extra config file will make poetry create the virtualenv in a folder called `.venv/` in the root directory, and this will be the case for anyone who pulls the repository.

File: bot/arena_bot.py [hunk 1]
```diff
@@ -8,8 +8,8 @@
 from discord import Intents
 from discord.ext.commands import Bot
 
-from config import COMMAND_PREFIX
-from database import Database
+from bot.config import COMMAND_PREFIX
+from bot.database import Database
```
Comment (GDWR) [inline bot/arena_bot.py:12]: Why have all the imports for source files been prefixed with `bot.`? I want to avoid this.

File: bot/arena_bot.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ async def on_disconnect(self) -> None:
     def load_extensions(self) -> None:
         """Iterate through all the `.py` files found int `bot/cogs` and load them as cogs."""
         for file in map(Path, glob("bot/cogs/*.py")):
-            self.load_extension(f"cogs.{file.stem}")
+            self.load_extension(f"bot.cogs.{file.stem}")
```
Comment (GDWR) [inline bot/arena_bot.py:49]: The working Directory should be the bot folder

**Review by GDWR — CHANGES_REQUESTED** [review]
I don't want to have to prefix all source files with `bot.` the working directory should be the bot folder.

File: bot/arena_bot.py (hunk 1 above)
Comment (DudeBro249) [reply bot/arena_bot.py]: Ive made the necessary changes

File: bot/arena_bot.py (hunk 2 above)
Comment (DudeBro249) [reply bot/arena_bot.py]: Done

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+[tool.poetry]
+name = "bot"
```
Comment (GDWR) [inline pyproject.toml:2]:
```suggestion
name = "DiscordArena"
```

**Review by GDWR — APPROVED** [review]
Update Poetry project name then merge

**Review by GDWR — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (DudeBro249) [reply pyproject.toml]: Done

**Pull request merged**